src/cargo/core/compiler/job_queue/mod.rs

```diff
@@ -1129,6 +1136,13 @@ impl<'cfg> DrainState<'cfg> {
                     if unit.mode.is_check() {
                         config.shell().status("Checking", &unit.pkg)?;
                     } else {
+                        if json {
+                            let msg = machine_message::CompileStarted {
+                                package_id: unit.pkg.package_id(),
```

Comment:
This should also include the target and platform, to allow distinguishing the different crates of a package and builds for the host vs target (vs artifact-dependencies targets).

There's also other unit types that "start", like `run-custom-build`, which would be useful to have a message for too.

Reply:
> include the target and platform

The other cargo messages dont do this, it would be pointless for this one to, unless all of them did?

Reply:
Would `run-custom-build` be the start of a `build-script-executed`?

I think I can do that in another PR.

Reply:
> The other cargo messages dont do this, it would be pointless for this one to, unless all of them did?

The others do include the `target`, IMO them not including the `platform` is a bug (https://github.com/rust-lang/cargo/issues/12869), but maybe best to see what the maintainers say.

> Would `run-custom-build` be the start of a `build-script-executed`?

Yep, exactly.

> I think I can do that in another PR.

:+1: